src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/entity/PlanEntity.kt

```diff
@@ -1,104 +1,80 @@
 package uk.gov.justice.digital.hmpps.sentenceplan.entity
 
 import com.fasterxml.jackson.annotation.JsonIgnore
-import jakarta.persistence.CascadeType
 import jakarta.persistence.Column
 import jakarta.persistence.Entity
-import jakarta.persistence.EntityListeners
 import jakarta.persistence.EnumType
 import jakarta.persistence.Enumerated
-import jakarta.persistence.FetchType
 import jakarta.persistence.GeneratedValue
 import jakarta.persistence.GenerationType
 import jakarta.persistence.Id
 import jakarta.persistence.JoinColumn
 import jakarta.persistence.ManyToOne
-import jakarta.persistence.OneToMany
 import jakarta.persistence.OneToOne
-import jakarta.persistence.OrderBy
 import jakarta.persistence.Table
 import jakarta.transaction.Transactional
+import org.springframework.data.annotation.CreatedBy
 import org.springframework.data.annotation.CreatedDate
 import org.springframework.data.annotation.LastModifiedBy
 import org.springframework.data.annotation.LastModifiedDate
-import org.springframework.data.jpa.domain.support.AuditingEntityListener
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.Modifying
 import org.springframework.data.jpa.repository.Query
 import org.springframework.data.repository.query.Param
 import java.time.Instant
 import java.util.UUID
 
-@Entity(name = "Plan")
+@Entity
 @Table(name = "plan")
-@EntityListeners(AuditingEntityListener::class)
 class PlanEntity(
   @Id
-  @Column(name = "id")
   @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "id", nullable = false)
   @JsonIgnore
-  val id: Long? = null,
+  var id: Long? = null,
 
-  @Column(name = "uuid")
-  val uuid: UUID = UUID.randomUUID(),
-
-  @Column(name = "countersigning_status")
+  @Column(name = "published_state")
   @Enumerated(EnumType.STRING)
-  val status: CountersigningStatus = CountersigningStatus.INCOMPLETE,
+  var publishedState: PublishState = PublishState.UNPUBLISHED,
 
-  @Column(name = "agreement_status")
-  @Enumerated(EnumType.STRING)
-  var agreementStatus: PlanStatus = PlanStatus.DRAFT,
+  @Column(name = "uuid", nullable = false)
+  var uuid: UUID = UUID.randomUUID(),
 
   @CreatedDate
-  @Column(name = "creation_date")
-  val creationDate: Instant = Instant.now(),
+  @Column(name = "created_date")
+  var createdDate: Instant? = Instant.now(),
```

Comment:
Annotate `@CreatedDate`, but also initialise?

Reply:
The annotation means the value it will be overwritten in the pre-persist, and I really hate having to address fields via `?` when I know the values will exist! Absolutely missed converting these to LocalDateTime though, will do that (and for all others).